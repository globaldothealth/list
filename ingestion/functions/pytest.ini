[pytest]
filterwarnings = ignore::DeprecationWarning:(b|m)oto.*: