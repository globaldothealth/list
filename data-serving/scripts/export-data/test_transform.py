import csv
import pytest
import json
from pprint import pprint
import transform as T

_DEEP_GET = [
    ({"x": {"y": {"z": 2}}}, "x.y.z", 2),
    ({"x": {"y": [1, 2]}}, "x.y", [1, 2]),
]

_ADDITIONAL_SOURCES = [
    ("[]", None),
    (
        '[{"sourceUrl": "http://foo.bar"}, {"sourceUrl": "http://bar.baz"}]',
        "http://foo.bar,http://bar.baz",
    ),
]

_TRAVEL = [
    {
        "dateRange": {"start": "2021-10-10T00:00:00Z", "end": "2021-10-12T00:00:00Z"},
        "location": {
            "geometry": {"latitude": 35, "longitude": -31},
            "administrativeAreaLevel1": "Port",
            "country": "Atlantis",
            "name": "Port of Atlantis",
        },
        "methods": "Ship",
    },
    {
        "dateRange": {"start": "2021-10-13T00:00:00Z", "end": "2021-10-15T00:00:00Z"},
        "location": {
            "geometry": {"latitude": 35, "longitude": -31},
            "administrativeAreaLevel1": "Coast",
            "country": "Atlantis",
            "name": "Coast",
        },
        "methods": "Raft",
    },
]

_TRAVEL_parsed = {
    "travelHistory.travel.dateRange.end": "2021-10-12,2021-10-15",
    "travelHistory.travel.dateRange.start": "2021-10-10,2021-10-13",
    "travelHistory.travel.location.administrativeAreaLevel1": "Port,Coast",
    "travelHistory.travel.location.country": "Atlantis,Atlantis",
    "travelHistory.travel.location.geometry.coordinates": "(35, -31),(35, -31)",
    "travelHistory.travel.location.name": "Port of Atlantis,Coast",
    "travelHistory.travel.methods": "Ship,Raft",
}

def _read_csv(fn):
    with open(fn) as f:
        c = csv.DictReader(f)
        return [row for row in c]

@pytest.mark.parametrize("dictionary,key,value", _DEEP_GET)
def test_deep_get(dictionary, key, value):
    assert T.deep_get(dictionary, key) == value


@pytest.mark.parametrize("sources,expected", _ADDITIONAL_SOURCES)
def test_convert_addl_sources(sources, expected):
    assert T.convert_addl_sources(sources) == expected


def test_convert_travel():
    assert T.convert_travel(json.dumps(_TRAVEL)) == _TRAVEL_parsed


def test_convert_row():
    output = [T.convert_row(r) for r in _read_csv("test_transform_mongoexport.csv")]
    with open("test_transform_expected.json") as f:
        expected = json.load(f)
    assert output == expected
