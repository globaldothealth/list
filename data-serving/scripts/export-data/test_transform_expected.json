[
  {
    "_id": "ObjectId(5acaca17627617232598ff09)",
    "caseReference.additionalSources": "",
    "caseReference.sourceEntryId": "",
    "caseReference.sourceId": "5f748db4f9a454707d54faaa",
    "caseReference.sourceUrl": "http://foo/bar.csv",
    "caseReference.uploadIds": "",
    "caseReference.verificationStatus": "VERIFIED",
    "demographics.ageRange.end": "",
    "demographics.ageRange.start": "",
    "demographics.ethnicity": "",
    "demographics.gender": "",
    "demographics.nationalities": "",
    "demographics.occupation": "",
    "events.confirmed.date": "2020-02-28",
    "events.confirmed.value": "",
    "events.firstClinicalConsultation.date": "",
    "events.firstClinicalConsultation.value": "",
    "events.hospitalAdmission.date": "",
    "events.hospitalAdmission.value": "",
    "events.icuAdmission.date": "",
    "events.icuAdmission.value": "",
    "events.onsetSymptoms.date": "",
    "events.onsetSymptoms.value": "",
    "events.outcome.date": "",
    "events.outcome.value": "",
    "events.selfIsolation.date": "",
    "events.selfIsolation.value": "",
    "genomeSequences": "",
    "genomeSequences.repositoryUrl": "",
    "genomeSequences.sampleCollectionDate": "",
    "genomeSequences.sequenceId": "",
    "genomeSequences.sequenceLength": "",
    "genomeSequences.sequenceName": "",
    "location.administrativeAreaLevel1": "Port",
    "location.administrativeAreaLevel2": "Oldport",
    "location.administrativeAreaLevel3": "",
    "location.country": "Atlantis",
    "location.geoResolution": "Admin2",
    "location.geometry.latitude": "35",
    "location.geometry.longitude": "-31",
    "location.name": "Port of Atlantis",
    "location.place": "",
    "notes": "",
    "pathogens": "",
    "preexistingConditions.hasPreexistingConditions": "",
    "preexistingConditions.values": "",
    "revisionMetadata.creationMetadata.date": "2021-05-13T12:40:40.000Z",
    "revisionMetadata.creationMetadata.notes": "",
    "revisionMetadata.editMetadata.date": "",
    "revisionMetadata.editMetadata.notes": "",
    "revisionMetadata.revisionNumber": "0",
    "symptoms.status": "",
    "symptoms.values": "",
    "transmission.linkedCaseIds": "",
    "transmission.places": "",
    "transmission.routes": "",
    "travelHistory.travel.dateRange.end": "",
    "travelHistory.travel.dateRange.start": "",
    "travelHistory.travel.location.administrativeAreaLevel1": "",
    "travelHistory.travel.location.administrativeAreaLevel2": "",
    "travelHistory.travel.location.administrativeAreaLevel3": "",
    "travelHistory.travel.location.country": "",
    "travelHistory.travel.location.geoResolution": "",
    "travelHistory.travel.location.geometry.coordinates": "",
    "travelHistory.travel.location.name": "",
    "travelHistory.travel.location.place": "",
    "travelHistory.travel.methods": "",
    "travelHistory.travel.purpose": "",
    "travelHistory.traveledPrior30Days": "",
    "vaccines.0.batch": "",
    "vaccines.0.date": "",
    "vaccines.0.name": "",
    "vaccines.0.sideEffects": "",
    "vaccines.1.batch": "",
    "vaccines.1.date": "",
    "vaccines.1.name": "",
    "vaccines.1.sideEffects": "",
    "vaccines.2.batch": "",
    "vaccines.2.date": "",
    "vaccines.2.name": "",
    "vaccines.2.sideEffects": "",
    "vaccines.3.batch": "",
    "vaccines.3.date": "",
    "vaccines.3.name": "",
    "vaccines.3.sideEffects": "",
    "variantOfConcern": ""
  },
  {
    "_id": "ObjectId(609d20396760e8456eb70c80)",
    "caseReference.additionalSources": "",
    "caseReference.sourceEntryId": "",
    "caseReference.sourceId": "5f748db4f9a454707d54faaa",
    "caseReference.sourceUrl": "http://foo/bar.csv",
    "caseReference.uploadIds": "",
    "caseReference.verificationStatus": "VERIFIED",
    "demographics.ageRange.end": "",
    "demographics.ageRange.start": "",
    "demographics.ethnicity": "",
    "demographics.gender": "",
    "demographics.nationalities": "",
    "demographics.occupation": "",
    "events.confirmed.date": "2020-03-31",
    "events.confirmed.value": "",
    "events.firstClinicalConsultation.date": "",
    "events.firstClinicalConsultation.value": "",
    "events.hospitalAdmission.date": "",
    "events.hospitalAdmission.value": "",
    "events.icuAdmission.date": "",
    "events.icuAdmission.value": "",
    "events.onsetSymptoms.date": "",
    "events.onsetSymptoms.value": "",
    "events.outcome.date": "",
    "events.outcome.value": "",
    "events.selfIsolation.date": "",
    "events.selfIsolation.value": "",
    "genomeSequences": "",
    "genomeSequences.repositoryUrl": "",
    "genomeSequences.sampleCollectionDate": "",
    "genomeSequences.sequenceId": "",
    "genomeSequences.sequenceLength": "",
    "genomeSequences.sequenceName": "",
    "location.administrativeAreaLevel1": "Coast",
    "location.administrativeAreaLevel2": "",
    "location.administrativeAreaLevel3": "",
    "location.country": "Atlantis",
    "location.geoResolution": "Admin1",
    "location.geometry.latitude": "35.2",
    "location.geometry.longitude": "-32",
    "location.name": "Coast, Atlantis",
    "location.place": "",
    "notes": "",
    "pathogens": "",
    "preexistingConditions.hasPreexistingConditions": "",
    "preexistingConditions.values": "",
    "revisionMetadata.creationMetadata.date": "2021-05-13T12:40:40.000Z",
    "revisionMetadata.creationMetadata.notes": "",
    "revisionMetadata.editMetadata.date": "",
    "revisionMetadata.editMetadata.notes": "",
    "revisionMetadata.revisionNumber": "0",
    "symptoms.status": "",
    "symptoms.values": "",
    "transmission.linkedCaseIds": "",
    "transmission.places": "",
    "transmission.routes": "",
    "travelHistory.travel.dateRange.end": "",
    "travelHistory.travel.dateRange.start": "",
    "travelHistory.travel.location.administrativeAreaLevel1": "",
    "travelHistory.travel.location.administrativeAreaLevel2": "",
    "travelHistory.travel.location.administrativeAreaLevel3": "",
    "travelHistory.travel.location.country": "",
    "travelHistory.travel.location.geoResolution": "",
    "travelHistory.travel.location.geometry.coordinates": "",
    "travelHistory.travel.location.name": "",
    "travelHistory.travel.location.place": "",
    "travelHistory.travel.methods": "",
    "travelHistory.travel.purpose": "",
    "travelHistory.traveledPrior30Days": "",
    "vaccines.0.batch": "",
    "vaccines.0.date": "",
    "vaccines.0.name": "",
    "vaccines.0.sideEffects": "",
    "vaccines.1.batch": "",
    "vaccines.1.date": "",
    "vaccines.1.name": "",
    "vaccines.1.sideEffects": "",
    "vaccines.2.batch": "",
    "vaccines.2.date": "",
    "vaccines.2.name": "",
    "vaccines.2.sideEffects": "",
    "vaccines.3.batch": "",
    "vaccines.3.date": "",
    "vaccines.3.name": "",
    "vaccines.3.sideEffects": "",
    "variantOfConcern": ""
  }
]